packages/media-signaling/src/definition/signals/client/hangup.ts

```diff
@@ -0,0 +1,50 @@
+import type { JSONSchemaType } from 'ajv';
+
+import type { CallHangupReason } from '../../call';
+
+/** Client is saying they hanged up from a call. The reason specifies if its a clean hangup or an error */
+export type ClientMediaSignalHangup = {
+	callId: string;
+	contractId: string;
+	type: 'hangup';
+
+	reason: CallHangupReason;
+};
+
+export const clientMediaSignalHangupSchema: JSONSchemaType<ClientMediaSignalHangup> = {
+	type: 'object',
+	properties: {
+		callId: {
+			type: 'string',
+			nullable: false,
+			minLength: 1,
+		},
+		contractId: {
+			type: 'string',
+			nullable: false,
+			minLength: 1,
+		},
+		type: {
+			type: 'string',
+			const: 'hangup',
+		},
+		reason: {
+			type: 'string',
+			enum: [
+				'normal',
+				'remote',
+				'rejected',
+				'unavailable',
+				'transfer',
+				'timeout',
+				'signaling-error',
+				'service-error',
+				'media-error',
+				'error',
+			],
+			nullable: false,
```

Comment:
iirc, if prop is required, `null` and `undefined` fail automatically

Reply:
this one actually doesn't need the nullable, but it's thanks to the enum.